framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -330,6 +333,106 @@ public void processClassTree(ClassTree classTree) {
         super.visitClass(classTree, null);
     }
 
+    /**
+     * Check that the field invariant meets the following requirements:
```

Comment:
That doesn't help a person to understand this comment; the person doesn't know that in order to define this particular term, he or she needs to read the manual section.
I also feel that a term like "refined field invariant" would be better.  `@FieldInvariant` is a fine name for the annotation, but "field invariant" doesn't suggest the fact that it is different than in the parent.  If someone mentioned a "field invariant" to me, I would think of just the annotation on the field.